### PR TITLE
Vz 5822 VMO metrics configuration

### PR DESCRIPTION
**Pull request by zabdulre**
# Description
Created a service monitor for the VMO metrics endpoint. Updated the bom to use the latest version of the VMO. Updated VMO deployment and service to ensure compatibility with serviceMonitor. Updated promoperator test to account for all 3 metrics-related serviceMonitors 
Fixes VZ-5822 *NOTE: Unit tests will fail since not all 3 metrics serviceMonitors have been merged with master yet. This PR will be merged last to avoid any issues
# Checklist
As the author of this PR, I have:
- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [x] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate
Code reviewer, please confirm this PR:
- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        - sourceLabels: 
+            - __meta_kubernetes_service_name 
+            - __meta_kubernetes_endpoint_port_name
+          regex: verrazzano-monitoring-operator;http-prom
+          action: keep
+        - sourceLabels:
+          - __meta_kubernetes_service_label_app 
+          targetLabel: app 
+      scheme: https                               #should not be added as a relable
+      path: /metrics                              #path can be added as a relable but it is then not shown in the prometheus UI
```
Comment (davisgomes) [inline platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml:27]: You can remove these comments

Comment (zabdulre) [reply platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml:27]: will do

File: platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      tlsConfig:
```
Comment (davisgomes) [inline platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml:28]: We may need to update this for the case where Istio is disabled.

File: .gitignore [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Copyright (c) 2020, 2022, Oracle and/or its affiliates.
 # Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 .verrazzano/
+build_from_jenkins.sh
```
Comment (mbenmclean) [inline .gitignore:4]: Why is this added here?

File: platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml [hunk 4]
```diff
@@ -31,6 +32,9 @@ spec:
             - containerPort: {{ .Values.monitoringOperator.metricsPort }}
               name: http-metrics
               protocol: TCP
+            - containerPort: {{ .Values.monitoringOperator.exporterPort }}
+              name: http-prom
```
Comment (mbenmclean) [inline platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml:36]: Is the http-metrics port the one used by Istio?

File: platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml (hunk 2 above)
Comment (mbenmclean) [reply platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml:28]: Do we support an install without istio yet? I know we've done POCs and prototypes but I think istio is still required at this point from a user configuration perspective

File: .gitignore (hunk 3 above)
Comment (zabdulre) [reply .gitignore:4]: this is just a script I had in my directory, I'll remove this and clean it up

File: platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml (hunk 4 above)
Comment (zabdulre) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml:36]: I believe that port is for the envoy metrics

Comment (mbenmclean) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml:36]: Ok that's what I thought. Also, I think I prefer the name http-exporter to http-prom, but your call

Comment (zabdulre) [reply platform-operator/helm_config/charts/verrazzano-monitoring-operator/templates/deployment.yaml:36]: for sure, I think i will leave as is since I made sure to pass all tests in prior commits with this configuration

File: platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml (hunk 2 above)
Comment (zabdulre) [reply platform-operator/thirdparty/manifests/prometheus-operator/vmo_monitor.yaml:28]: Is this something we can revisit in a later PR?

**Review by davisgomes — APPROVED** [review]
